### PR TITLE
log-backup: applied some change to make better RPO

### LINKED ISSUE #13941: Reduce the lag of log backup during rolling updating
## Development Task
As title

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #13941 Ref https://github.com/pingcap/tidb/issues/39620

What's Changed:
- If failed to get initial snapshot, remove the subscription as soon as possible. *(Given there isn't any possibility this subscription can be resumed.)*
- Added a cache of getting checkpoint. This cache is lease-based -- the lease time is simply the tick interval of the coordinator. *(Which can greatly speedup the initial scanning when there are many regions -- getting the checkpoint may take ~50ms and typical initial scanning takes only ~5ms)*
- Make the channel size huger for don't blocking the main loop when many regions migrating.

```commit-message
- If failed to get initial snapshot, remove the subscription as soon as possible. 
- Added a cache of getting checkpoint. This cache is lease-based -- the lease time is simply the tick interval of the coordinator. 
- Make the channel size huger for don't blocking the main loop when many regions migrating.

```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
This is the lag when applied this PR and changed flush interval to 1 min.
image
While this is the lag of master when changed flush interval to 1 min.
image

### Release note 

```release-note
Log Backup now has smaller lag when doing rolling update.
```

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: components/backup-stream/src/metadata/client.rs [hunk 1]
```diff
@@ -698,21 +702,41 @@ impl<Store: MetaStore> MetadataClient<Store> {
...
+            })
+    }
+
+    fn update_cache(&self, task: &str, checkpoint: TimeStamp) {
+        let mut c = self.caches.entry(task.to_owned()).or_default();
+        c.value_mut().update(checkpoint);
+    }
+
     pub async fn get_region_checkpoint(&self, task: &str, region: &Region) -> Result<Checkpoint> {
+        if let Some(c) = self.cached_checkpoint(task) {
```
Comment (joccau) [inline components/backup-stream/src/metadata/client.rs:721]: Whether we need to consider that the checkpoint from PD is larger than cached checkpoint in local TiKV ?

Comment (YuJuncen) [reply components/backup-stream/src/metadata/client.rs:721]: This is what the "lease" did, the lease time is 12s (which is the same as TiDB's tick interval), the checkpoint probably won't be increased in this interval.

**Review by joccau — APPROVED** [review]
LGTM

File: components/backup-stream/src/metadata/client.rs (hunk 1 above)
Comment (YuJuncen) [reply components/backup-stream/src/metadata/client.rs:721]: (In fact, perhaps we can reduce the lease time in the future)

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]

**Pull request merged**